src/js/components-react/DateTime.js

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DateTimePicker, {registerLocale} from 'react-datepicker';
+import ja from 'date-fns/locale/ja';
+import zh from 'date-fns/locale/zh-CN';
+import en from 'date-fns/locale/en-US';
+
+class DateTime extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    ja.options.firstWeekContainsDate = 1;
+    ja.options.weekStartsOn = 0;
+    en.options.firstWeekContainsDate = 1;
+    en.options.weekStartsOn = 0;
+    zh.options.firstWeekContainsDate = 1;
+    zh.options.weekStartsOn = 0;
+    registerLocale('ja', ja);
+    registerLocale('zh', zh);
+    registerLocale('en', en);
+  }
+  _handleChangeDate = (date) => {
+    if (this.props.value === '') {
+      this.props.value = date;
```

Comment:
Is there any special reason to assign to this.props.value?
It is generally a bad idea to modify this.props passed from parent scope.

Reply:
I thought this is the only way to control props.
But Is there a better way to control props? If it is, please tell me.

Reply:
this.props is what the component receives from parent scope. So any modification in ideal should be modified in parent scope through state. It could look something like this:
...
class App extends React.component {
   state={
     // here in parent scope we have a state that represents the selected date on our DateTime component
      selectedDateTime: Date()
   };
   _handleOnChange(newDate) {
     // each time user selects new date set the selectedDateTime state to the new date so the re-render can occur
      this.setState({selectedDateTime: newDate});
   }
   render() {
      // here props.value of our DateTime component is assigned to a state in parent scope App, so whenever the state changes DateTime component will start its own render cycle too
      <DateTime value={this.state.selectedDateTime} onChange={this._handleOnChange} />
   }
}

Please, ask if there is anything not clear above.
Thank you!